client/src/main/java/com/arcbees/beeshop/client/application/ApplicationModule.java

```diff
@@ -28,6 +28,10 @@ protected void configure() {
         install(new ProductModule());
         install(new WidgetModule());
 
+        bind(BrandChangeHandler.class).asEagerSingleton();
+        bind(CurrentBrand.class).to(CurrentBrandImpl.class);
+        bind(ThemeChanger.class).to(ThemeChangerImpl.class).asEagerSingleton();
```

Comment:
Another way of doing that without an extra eager singleton would be to have a Proxy Event on the root presenter (The ApplicationPresenter here).